components/director/internal/tenantfetchersvc/resync/client.go

```diff
@@ -187,13 +174,8 @@ func NewClient(oAuth2Config OAuth2Config, authMode oauth.AuthMode, clientConfig
 	}, nil
 }
 
-// SetMetricsPusher missing godoc
-func (c *Client) SetMetricsPusher(metricsPusher MetricsPusher) {
-	c.metricsPusher = metricsPusher
-}
-
 // FetchTenantEventsPage missing godoc
-func (c *Client) FetchTenantEventsPage(eventsType EventsType, additionalQueryParams QueryParams) (*EventsPage, error) {
+func (c *Client) FetchTenantEventsPage(ctx context.Context, eventsType EventsType, additionalQueryParams QueryParams) (*EventsPage, error) {
```

Comment:
why is context needed here?

Reply:
because of the logs in the metrics pusher - they went from `log.D()` to `log.C(ctx)`